lib/checkother.cpp

```diff
@@ -1225,6 +1225,16 @@ bool CheckOther::checkInnerScope(const Token *tok, const Variable* var, bool& us
                         if (!ret.empty() && ret.back() == '*')
                             return false;
                     }
+                    if (ftok->function()) {
+                        const std::vector<const Token *> argtoks = getArguments(ftok);
+                        int argi = 0;
+                        for (const auto &argtok : argtoks) {
+                            // if theres is another reference argument, assume it depends on the current argument
+                            if (argi != argn && (Token::simpleMatch(argtok, "&") || ftok->function()->getArgumentVar(argi)->isReference()))
```

Comment:
I think `getArgumentVar()` can return `nullptr` for an unnamed parameter.

Reply:
Is it possible to check if it's a reference anyway?

Reply:
My comment was wrong, the variable for an unnamed parameter exists, but has no nameToken.
The check for a reference seems OK.